libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -152,6 +152,7 @@ class eos_vm_instantiated_module : public wasm_instantiated_module_interface {
          };
 
          execute(context, bkend, exec_ctx, wasm_alloc, fn, true);
+         context.control.return_sync_call_wasm_allocator(); // return the wasm_allocator obtainded by get_sync_wasm_allocator back to the pool
```

Comment:
Does this need to be `make_scoped` so it still runs upon exception?

Reply:
alternatively maybe `sync_call_wasm_alloc_index` can just be reset to 0 upon on a new apply context

Reply:
We cannot just reset to 0 upon on a new apply context, as the apply context can make multiple sequential sync calls. I changed to use scoped exit mechanism.

https://github.com/AntelopeIO/spring/pull/1251/commits/06bdd12b595735e7a885e3422925037f5d58eaa2

Reply:
I added a test to make 500 sync calls in a loop to verify wasm allocators are properly released after each call (we are not running out of them).

https://github.com/AntelopeIO/spring/pull/1251/commits/b1cbe35ce8114475972aebf0b994af7b1da39db6